File: app/src/main/java/com/github/android/lvrn/lvrnproject/persistent/entity/Entity.java
package com.github.android.lvrn.lvrnproject.persistent.entity;

import android.os.Parcelable;
import android.support.annotation.NonNull;

/**
 * @author Vadim Boitsov <vadimboitsov1@gmail.com>
 */

public abstract class Entity implements Parcelable {

    @NonNull
    protected String id;

    @NonNull
    public String getId() {
        return id;
    }

    public void setId(@NonNull String id) {
        this.id = id;
    }

    @Override
    public String toString() {
        return "Entity{" +
                "id='" + id + '\'' +
                '}';
    }
}

File: app/src/main/java/com/github/valhallalabs/laverna/persistent/entity/base/Entity.kt
package com.github.valhallalabs.laverna.persistent.entity.base

import android.os.Parcelable

/**
 * @author Vadim Boitsov <vadimboitsov1@gmail.com>
 * @author Oleksandr Tyshkovets <olexandr.tyshkovets@gmail.com>
 */
abstract class Entity : Parcelable {
    abstract val id: String

    override fun toString(): String {
        return "Entity{" +
                "id='" + id +
                '}'
    }

}
